Framework/Algorithms/src/ConvertUnitsUsingDetectorTable.cpp

```diff
@@ -147,7 +148,7 @@ MatrixWorkspace_sptr ConvertUnitsUsingDetectorTable::convertViaTOF(
 
   // Perform Sanity Validation before creating workspace
   size_t checkIndex = 0;
-  int checkSpecNo = inputWS->getDetector(checkIndex)->getID();
+  int checkSpecNo = inputWS->getSpectrum(checkIndex).getSpectrumNo();
```

Comment:
Doesn't this change the behavior of the algorithm? Does the column really contain spectrum numbers, or is the name `"spectra"` misleading?